server/images.go

```diff
@@ -344,24 +344,33 @@ func CreateModel(ctx context.Context, name, modelFileDir string, commands []pars
 			}
 			defer bin.Close()
 
-			fn(api.ProgressResponse{Status: "creating model layer"})
-			ggml, err := llm.DecodeGGML(bin)
-			if err != nil {
-				return err
-			}
+			var offset int64
+			for {
+				fn(api.ProgressResponse{Status: "creating model layer"})
+
+				bin.Seek(offset, io.SeekStart)
+				ggml, err := llm.DecodeGGML(bin)
+				if errors.Is(err, io.EOF) {
```

Comment:
What makes an EOF ok here? This is quite hidden and feels difficult to track down in the case it causes a bug.

Reply:
EOF indicates the model has been fully consumed by the previous iteration. it's a pattern that's no dissimilar to how io.Reader works